### PR TITLE
[vfs] change addon interface to new C++ style

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
Next addon system change, where VFS becomes changed to C++ style.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Related addon is visible here https://github.com/AlwinEsch/vfs.rar/tree/kodi-agile

File: xbmc/addons/VFSEntry.h [hunk 1]
```diff
@@ -89,8 +99,6 @@ namespace ADDON
     AddonInstance_VFSEntry m_struct; //!< VFS callback table
   };
 
-  typedef std::shared_ptr<CVFSEntry> VFSEntryPtr; //!< Convenience typedef.
-
   //! \brief Wrapper equpping a CVFSEntry with an IFile interface.
```
Comment (ksooo) [inline xbmc/addons/VFSEntry.h]: equpping? => "equipping" (there is an "i" missing)

**Review by ksooo — COMMENTED** [review]
not sure what the benefit of the new vfs addon cache is

File: xbmc/Application.cpp [hunk 2]
```diff
@@ -5219,14 +5208,8 @@ void CApplication::CloseNetworkShares()
...
 
-  VECADDONS addons;
-  CServiceBroker::GetBinaryAddonCache().GetAddons(addons, ADDON_VFS);
-  for (auto& it : addons)
-  {
-    AddonPtr addon = CServiceBroker::GetBinaryAddonCache().GetAddonInstance(it->ID(), ADDON_VFS);
-    VFSEntryPtr vfs = std::static_pointer_cast<CVFSEntry>(addon);
-    vfs->DisconnectAll();
-  }
+  for (const auto& vfsAddon : CServiceBroker::GetVFSAddonCache().GetAddonInstances())
```
Comment (AlwinEsch) [inline xbmc/Application.cpp]: The binary addon cache becomes complete removed soon, the vfs addon is the only one where the related cache parts becomes not handled by a own manager.

Comment (ksooo) [inline xbmc/Application.cpp]: ok, then

File: xbmc/addons/VFSEntry.cpp [hunk 3]
```diff
@@ -483,17 +526,19 @@ bool CVFSEntryIDirectoryWrapper::GetDirectory(const CURL& url,
 
 bool CVFSEntryIDirectoryWrapper::DoGetKeyboardInput(void* ctx,
                                                     const char* heading,
-                                                    char** input)
+                                                    char** input,
+                                                    bool hidden_input)
```
Comment (AlwinEsch) [inline xbmc/addons/VFSEntry.cpp]: Thought to do it here, the way how the Keyboard on addon becomes used is complete changed.

How it works has also confused me during test and added direct a documentation part here: https://github.com/AlwinEsch/kodi/blob/0a380b04591da8a99b2fc56c68fa7b3c8f6e3cf6/xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/VFS.h#L278-L330

**Comment by AlwinEsch** [comment]
Is updated

**Review by ksooo — APPROVED** [review]

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**